feature/KAS-3108 Pub v3: Details: /publicatication - published-decisions panel

mdv-professional opened the pull request:
Deze branch is bij lange nog niet klaar. Ik heb een aantal vragen. Ik weet van mezelf soms moeite te hebben met inschatten waar consistentie vs. custom aanpak gewenst is.

Commit:
(copy) start from publication/decisions/index route
KAS-3108 pub: v3: details: publications

Commit:
add subroute publications
KAS-3108 pub: v3: details: publications

Commit:
show very basic timeline
KAS-3108 pub: v3: details: publications

Commit:
add basic panels to timeline
KAS-3108 pub: v3: details: publications

mdv-professional commented on app/components/utils/generic/timeline/timeline.hbs:
```diff
@@ -0,0 +1,9 @@
+<div>
```
Een tijdslijn met acties komt voor onder /vertalingen /drukproeven & /publicaties. Is deze component zinvol?

erikap replied:
Die component lijkt me zeker zinvol, maar ik zou hem wel een level hoger definiëren zodat het `<Utils::Generic::Timeline>` is ipv `<Utils::Generic::Timeline::Timeline>` (de timeline-event kan wel onder `utils/generic/timeline/timeline-event` blijven staan)

mdv-professional replied:
Ik wist niet goed of ik die zou als "is nodig" moest interpreteren. Voorlopig heb ik het laten zitten. Ik kan daar beter aan uit.

mdv-professional commented on app/pods/publications/publication/publication-activities/index/controller.js:
```diff
@@ -0,0 +1,118 @@
+import { action } from '@ember/object';
+import { tracked } from '@glimmer/tracking';
+import { dropTask, task } from 'ember-concurrency';
+/* eslint-disable no-unused-vars */
```
Dit helpt m'n vscode. Ik verwijder het wanneer ik klaar ben indien gewenst.

mdv-professional commented on app/pods/publications/publication/publication-activities/index/route.js:
```diff
@@ -0,0 +1,37 @@
+import Route from '@ember/routing/route';
```
Ik heb gezien dat een aparte route in de "Vertalingen"-pull-request werd gesuggereerd. Ik vind dit een verwarrende aanpak. Ik zou dit liever vermijden.

erikap replied:
Wat vind je daar precies verwarrend aan? Door de aparte route wordt het model/logica die nodig  is voor het info-panel  gescheiden van het model/logica die nodig is voor de timeline, wat me duidelijker lijkt?

mdv-professional replied:
Ik had het pattern nog nooit gezien. Ik begrijp nu wel de logica.

mdv-professional commented on app/pods/publications/publication/publication-activities/index/controller.js:
```diff
@@ -0,0 +1,118 @@
+  return publicationTimeline;
+}
+
+export class PublicationRequestEvent {
```
Ik heb hier gekozen voor row-objecten, zoals dat was in de oude proofs-route. Ik heb de "Vertalingen"-pull-request gedeeltelijk doorgenomen. Deze is nog niet af, maar heb gezien dat Tanee hier voor extra components heeft gekozen. Dit kan ik ook doen (Al zou ik de row-objecten behouden om ze op datum te sorteren.)

erikap replied:
Als ik je intentie goed begrijp is het hier de bedoeling om objecten te maken die dezelfde 'interface', nl. `title` en `date`, aanbieden, om zo in de template makkelijker gewoon over de lijst te kunnen loopen en weergeven?

Ik vind de static `create` methods wat verwarrend. Die lijken voor mij te impliceren dat er speciale dingen moeten gebeuren die niet met een gewone constructor kunnen, maar als je er dan naar gaat kijken, is dat niet het geval (of mis ik iets?).

Kunnen we de 2 aparte classes misschien mergen naar 1 class `PublicationEvent` met een `@tracked model` dat zowel een `publication-activity` als een `decision` kan zijn? Op die class dan `isRequest` en `isPublication` als get-function definieren die op basis van `this.args.model.constructor.modelName` de correcte boolean teruggeeft. En in de get-functions van `title` en `date` dan met `if (this.isRequest) { ... } else if (this.isPublication) { ... }` de juiste waarde teruggeven. Dat maakt voor mij alleszins de intentie van die class iets beter duidelijk.

mdv-professional replied:
- [x] de `create`-method is weggewerkt.
- [ ] Ik heb de opdeling volgens model-type wel behouden. Het doel van deze klasse is net om om te kunnnen met de verschillen tussen `request-activity` en `publication-activity`.

mdv-professional commented on app/pods/publications/publication/publication-activities/index/template.hbs:
```diff
@@ -0,0 +1,79 @@
+        {{#if event.isRequest}}
+          {{!Aanvraag}}
+          <Auk::Panel>
+            <Auk::Panel::Header>
```
Hier zou een `Auk::AccordionPanel` moeten komen. De huidige versie daarvan laat niet toe dat een titel meer is dan een string. Is het OK dat ik hier de Auk-component aanpas om naast een @title-attribute ook een :title-block te aanvaarden?

brenner-company replied:
@mdv-professional Ik heb van @TaneeG vorige week quasi dezelfde vraag gekregen. Binnen `feature/KAS-3106_Vertalingen_herwerken` (#1193) heb ik daarvoor `accordion-panel` en `panel` uitgebreid met een `@subtitle` argument.

![CleanShot 2022-02-14 at 11 11 15@2x](https://user-images.githubusercontent.com/18174827/153844745-1b24ecda-b08c-4d36-bfdb-c4f5ab3910c8.png)

Wat lijkt je het meest makkelijk om die changes bij te krijgen? Wacht je tot de branch gemerged is naar `development` of geef ik de nodige commits door zodat je ze desnoods kan cherry-picken?

mdv-professional replied:
OK. Ik heb ze gekopieerd.

erikap commented on app/pods/publications/publication/publication-activities/index/route.js:
```diff
@@ -0,0 +1,37 @@
+import { inject as service } from '@ember/service';
+import { createTimeline } from './controller';
+
+export default class PublicationsPublicationDecisionsIndexRoute extends Route {
```
Vergeet de naam van de class niet te wijzigen

erikap commented on app/pods/publications/publication/publication-activities/route.js:
```diff
@@ -1,12 +1,17 @@
 import Route from '@ember/routing/route';
```
Hier ben ik niet helemaal mee met je vraag. In de code die je nu aan het maken bent, zal de model-hook van `publications.publication.publication-activities.index` toch `publication-activities` als model bevatten?
Initieel noemde dit `publication` maar ik heb toen gevraagd om dat te wijzigen zodat het onderscheid met de parent-routes, die ook `publications` en `publication` noemen, duidelijker is. Wat de vertaling in de URL is, kan me minder schelen. Dat mag je wijzigen naar `publicatie` van mij.

Commit:
update Auk::Panel and Auk::AccordionPanel
KAS-3108 pub: v3: details: publications

Commit:
improve timeline
KAS-3108 pub: v3: details: publications: published decisions

Commit:
move timeline event panels to components
KAS-3108 pub: v3: details: publications

Commit:
add necessary relationships to include parameter
KAS-3108 pub: v3: details: publications

Commit:
(todo)
KAS-3108 pub: v3: details: publications

Commit:
(layout) restore no whitespace in email body
KAS-3108 pub: v3: details: publications

Commit:
(style) use `auk-u-text-pre-line` for e-mail body
KAS-3108 pub: v3: details: publications

Commit:
(name)
KAS-3108 pub: v3: details: publications

Commit:
(move) publication-timeline-event-panels
KAS-3108 pub: v3: details: publications

Commit:
(name) publication-timeline-event-panels
KAS-3108 pub: v3: details: publications

Commit:
(move) publication-timeline-event-panels (2)
KAS-3108 pub: v3: details: publications

Commit:
(name) publication-timeline-event-panels (2)
KAS-3108 pub: v3: details: publications

Commit:
very basic PublicationRequestModal
KAS-3108 pub: v3: details: publications

Commit:
very basic UploadRequestModal
KAS-3108 pub: v3: details: publications

Commit:
move model hook down
KAS-3108 pub: v3: details: publications

Commit:
(todo)
KAS-3108 pub: v3: details: publications

Commit:
validations / PublicationRequestModal
KAS-3108 pub: v3: details: publications

Commit:
(refactor) put publicationRequestEmail to new method
KAS-3108 pub: v3: details: publications

Commit:
fill in email / PublicationRequestModal
KAS-3108 pub: v3: details: publications

Commit:
(cleanup)
KAS-3108 pub: v3: details: publications

Commit:
(comment) clarify checks + cleanup method in cancel task / PublicationRequestModal
KAS-3108 pub: v3: details: publications

Commit:
(deprecation-fix) use `file.filename` attribute / PublicationRequestModal
KAS-3108 pub: v3: details: publications

Commit:
move `requestPublication` to `PublicationService`
KAS-3108 pub: v3: details: publications

Commit:
Merge branch 'development' into feature/KAS-3108-pub-v3-details--publications

Commit:
(code-style) const
KAS-3108 pub: v3: details: publications

Commit:
Merge branch 'feature/KAS-3108-pub-v3-details--publications' of github.com:kanselarij-vlaanderen/frontend-kaleidos into feature/KAS-3108-pub-v3-details--publications

Commit:
(cleanup) unused `isViaCouncilOfMinisters` / /publications/:id/publication
KAS-3108 pub: v3: details: publications

Commit:
(name) correct route name / /publications/:id/publication
KAS-3108 pub: v3: details: publications

Commit:
🚧
KAS-3108 pub: v3: details: publications

Commit:
Merge branch 'development' into feature/KAS-3108-pub-v3-details--publications--copy

Commit:
consistent with translations: timeline model
KAS-3108 pub: v3: details: publications

Commit:
consistent with translations: sort order
KAS-3108 pub: v3: details: publications

Commit:
sort order of linked request and publication-activity
KAS-3108 pub: v3: details: publications

Commit:
(cleanup) remove decisions route leftover / /publications/:id/publication
KAS-3108 pub: v3: details: publications

Commit:
🚧
KAS-3108 pub: v3: details: publications

Commit:
🚧 consistent* with translations: uploads / PublicationRequestModal
*`document-containers` are saved in callback
KAS-3108 pub: v3: details: publications

Commit:
Merge branch 'feature/KAS-3108-pub-v3-details--publications' of github.com:kanselarij-vlaanderen/frontend-kaleidos into feature/KAS-3108-pub-v3-details--publications

Commit:
🚧 consistent* with translations: uploads / PublicationRequestModal
*`document-containers` are saved in callback
KAS-3108 pub: v3: details: publications

Commit:
consistent with translations: uploads* / PublicationRequestModal
*`document-containers` are saved in callback
KAS-3108 pub: v3: details: publications

Commit:
Use old document list*
* not used `Utils::Documents::List`
reasons:
    - "Versions" tab does not work when `piece` does not have a `save`d `document-container`
    - `@piece.file` did unnecessarily refetch file
KAS-3108 pub: v3: details: publications

Commit:
(move) publication-activities/... > publication-activities/index/...
KAS-3108 pub: v3: details: publications

Commit:
(move update) publication-activities/... > publication-activities/index/...
KAS-3108 pub: v3: details: publications

Commit:
(name) better component name / PublicationRequestModal
KAS-3108 pub: v3: details: publications

Commit:
(cleanup) remove empty component controllers / `Publications...EventPanel`s
KAS-3108 pub: v3: details: publications

Commit:
(lint-fix)
KAS-3108 pub: v3: details: publications

Commit:
Merge branch 'development' into feature/KAS-3108-pub-v3-details--publications

Commit:
(comment) tab + new line clarification
KAS-3108 pub: v3: details: publications

Commit:
(refactor) use a custom component for the whole `Utils::Generic::Timeline::Timeline` for a publication activity
<=> instead of yielded `Component` parameter for `Utils::Generic::Timeline::Timeline`
reason:
     - simpler .hbs flow
     - consistency with translations
KAS-3108 pub: v3: details: publications

Commit:
(refactor) more Kaleidos conforming component API Utils::Generic::Timeline::Timeline
KAS-3108 pub: v3: details: publications

Commit:
remove broken type imports
KAS-3108 pub: v3: details: publications

Commit:
small design corrections / /publications/:id/publication
KAS-3108 pub: v3: details: publications

Commit:
(move) timeline event components
KAS-3108 pub: v3: details: publications

Commit:
(name) timeline event components
KAS-3108 pub: v3: details: publications

Commit:
(cleanup)
KAS-3108 pub: v3: details: publications

Commit:
🚧 register modal
KAS-3108 pub: v3: details: publications

Commit:
register modal: hide unfinished publication-activities
KAS-3108 pub: v3: details: publications

Commit:
delete modal / publications/:id/publications
KAS-3108 pub: v3: details: publications

Commit:
fix close PublicationRegistrationModal on save / publications/:id/publications
KAS-3108 pub: v3: details: publications

Commit:
fix delete request-activity request / /publications/:id/publications
KAS-3108 pub: v3: details: publications

Commit:
Merge branch 'development' into feature/KAS-3108-pub-v3-details--publications

Commit:
fix name isRegistrationDisabled / /publications/:id/publication
KAS-3108 pub: v3: details: publications

Commit:
correct: remove title iput / /publications/:id/publication
KAS-3108 pub: v3: details: publications

Commit:
minor improvements / PublicationRegistrationModal
KAS-3108 pub: v3: details: publications

Commit:
🚧 edit publication / /publications/:id/publication
KAS-3108 pub: v3: details: publications

Commit:
🚧 allow edit of publication / /publications/:id/publication
KAS-3108 pub: v3: details: publications

Commit:
limit number of publication-request and number of decisions to 1 / /publications/:id/publication
KAS-3108 pub: v3: details: publications

Commit:
enable/disable edit and delete / /publications/:id/publication
KAS-3108 pub: v3: details: publications

Commit:
visual polishing / /publications/:id/publication
KAS-3108 pub: v3: details: publications

Commit:
Merge branch 'feature/KAS-3108-pub-v3-details--publications' of github.com:kanselarij-vlaanderen/frontend-kaleidos into feature/KAS-3108-pub-v3-details--publications

Commit:
fix non-generic timeline empty message
KAS-3108 pub: v3: details: publications

Commit:
publication translation texts
KAS-3108 pub: v3: details: publications

Commit:
publication translation texts
KAS-3108 pub: v3: details: publications

Commit:
fix: do not fetch proofing-activities / /publications/:id/publication
KAS-3108 pub: v3: details: publications

Commit:
fix: static title for events / /publications/:id/publication
KAS-3108 pub: v3: details: publications

Commit:
fix: show end date / PublicationsPublicationTimelineEvent
KAS-3108 pub: v3: details: publications

Commit:
allow email absent for legacy data / PublicationsRequestTimelineEvent
KAS-3108 pub: v3: details: publications

Commit:
Merge branch 'development' into feature/KAS-3108-pub-v3-details--publications

ghost commented:
Nog enkele kleine aanvullingen gemaakt, maar ik stop ermee voor dit weekend.
Resterende `// TODO`'s:
- [ ] gelijktrekken van status-update naar "Gepubliceerd" met publication-status-dropdown-pill
- [x] delete-mogelijkheid in upload
- [ ] samenvoegen van `PublicationService.createProofRequestActivity`-en `PublicationService.requestPublication` [UPDATE 2022/02/28]
- [ ] Publicatie-datum wijzigingen doorvoeren in andere models [UPDATE 2022/02/28]
- [x] Wanneer dit gebeurd is, ben ik van plan alles nog eens door te nemen.

De onopgeloste issues zullen in een andere branch aangepakt worden. [UPDATE 2022/02/28]

Commit:
code quality improvements
KAS-3108 pub: v3: details: publications

Commit:
Merge branch 'development' into feature/KAS-3108-pub-v3-details--publications

Commit:
code quality improvements: route
KAS-3108 pub: v3: details: publications

Commit:
(move) remove index/ component folder / /publicaties/:id/publicatie
reason: consistency with "Vertalingen" en "Drukproeven"
KAS-3108 pub: v3: details: publications

Commit:
update component invocations after move / /publicaties/:id/publicatie
KAS-3108 pub: v3: details: publications

Commit:
code quality improvements: components
KAS-3108 pub: v3: details: publications

Commit:
(move) PublicationRegistrationModal > PublicationDetailsModal
KAS-3108 pub: v3: details: publications

Commit:
(rename) PublicationRegistrationModal > PublicationDetailsModal
KAS-3108 pub: v3: details: publications

Commit:
(lint-fix) remove new line
KAS-3108 pub: v3: details: publications

erikap committed:
Add page title to publications tab

erikap committed:
Remove type definitions and unused imports

erikap committed:
Streamline proofing and publication timeline models

erikap committed:
Rename timeline and timeline-event components

erikap committed:
Refactor timeline components and use on all publication tabs

- Remove timeline component: no added value, only additional
complexity
- Rename timeline event component
- Use component for translation/proofing timeline as well

erikap committed:
Rename publication components to be in line with translation/proofing components

erikap committed:
Split publication registration and edit modals

erikap committed:
Streamline publication request modal with proof/translation request modals

erikap committed:
Streamline publication requested panel with translation/proof
requested panel

erikap committed:
Merge requested-panel components into generic request-activity panel component

erikap committed:
Use toolbar contextual components on publication timeline pages

erikap committed:
Fix missing translation

erikap committed:
Fix linitng
